### PR TITLE
Add hda support

### PR DESCRIPTION
**Pull request by manuelkoester**
Hey!

Thanks again for this tool! It's very well done :)

With this PR I added support for diffing HDAs and changed some minor data structures. Some naming is tied very much to hip files, but I tried my best to keep as close to the existing naming as possible.

This is a screenshot of it in action with the supplied test HDAs.
image

Let me know if there's anything you would like to see changed/added.

Cheers,
Manuel

### CODE REVIEW

File: ui/custom_standart_item_model.py [hunk 1]
```diff
@@ -128,10 +129,9 @@ def populate_with_data(self, data, view_name: str) -> None:
                 )
                 parent_path = node_data.parent_path
                 parent_item = self.get_item_by_path(parent_path)
-                if parent_item or parent_path == "/":
```
Comment (manuelkoester) [inline ui/custom_standart_item_model.py:131]: I'm not entirely sure if it is save to remove this, but all unittests and the manual testing went fine.
This allows the Hdas to be created without a parent.

**Comment by golubevcg** [comment]
Hey Manuel!

Thank you a lot; you did a fantastic job!

I will be able to test this only on the weekend. When it's done, I will send you a message; it looks very nice!

At the moment, my unit test coverage can provide some testing, but it is far from good coverage - it does not cover UI at all, only API and even API is not at its full, so:)

Can I ask you, will your HDA diff also include differences in onCreation, onLoad and some node UI scripts (like disable when conditions etc.)?

**Comment by golubevcg** [comment]
Hello!
I checked your PR, and I really like what you did - the code is much better than in my branch.

The only problem I see is that it has only partial support for HDAs because any additional UI tweaks or script changes (e.g. onCreate and onLoad) will be ignored by this diff tool.

How can we fix this?
In Houdini API, you query HDA data using Python, this way, we can compare two HDA files, including all the Python and UI scripts; here is a little example:


~~~
import hou

hou.hipFile.load(
            hip_path, 
            suppress_save_prompt=True, 
            ignore_load_warnings=True
        )

# Get the node representing the HDA
hda = hou.node('/obj/box1/filecache1')

# Get the HDA definition
hda_def = hda.type().definition()

# Get the sections
sections = hda_def.sections()

# Iterate over the sections and print their names and content
for section_name, section in sections.items():
    print("Section Name:", section_name)
    print("Content:", section.contents())
    print("-" * 40)  # Print a separator

on_create_script = sections.get('OnCreated', None)
print("on_create_script:", on_create_script)

on_load_script = sections.get('OnLoaded', None)
print("on_load_script:", on_load_script)
~~~

This way, we can add a new item to our diff tool in case of any difference so the user can see that.

**Comment by golubevcg** [comment]
We can specify the settings icon for each section of the HDA.

If you don't have time to do that - no worries. Can you then create a PR into develop so I can merge your PR and finish this on my own:) This way, your contribution will be visible in our commits tree.

**Comment by manuelkoester** [comment]
Hey!
Sorry I was busy until now, but I actually have some additional WIP stuff for HDAs. I'll open up another PR which merges to develop and add the sections related WIP code.
Another thing that would probably be required is some dropdown to chose what definition of the HDA you actually want to compare..
Maybe we can work on the additional features together :P

**Comment by golubevcg** [comment]
Sounds amazing, no worries!

**Comment by golubevcg** [comment]
I will cancel this PR then, when you will finish, just open another to develop, so it will be easier for us to track all the changes.

**Pull request closed without merging**